.drone.star

```diff
@@ -313,3 +319,153 @@ def appBuilds(ctx):
+        {
+            "name": "ocis",
+            "image": "owncloud/ocis-rolling:master",
+            "detach": True,
+            "environment": environment,
+            "commands": [
+                "mkdir -p /var/lib/ocis/web/apps",
+                "cp support/drone/ocis.web.config.json /var/lib/ocis/web/config.json",
+                "cp dev/docker/ocis.apps.yaml /var/lib/ocis/apps.yaml",
+                "cp -r /apps/* /var/lib/ocis/web/apps/",
+                "ocis init || true && ocis server",
+            ],
+            "volumes": [
+                {
+                    "name": "apps",
+                    "path": "/apps",
+                },
+            ],
+        },
+    ]
+
+    wait_for_ocis = [
+        {
+            "name": "wait-for-ocis",
+            "image": OC_CI_ALPINE,
+            "commands": [
+                "timeout 200 bash -c 'while [ $(curl -sk -uadmin:admin " +
+                "%s/graph/v1.0/users/admin " % environment["OCIS_URL"] +
+                "-w %{http_code} -o /dev/null) != 200 ]; do sleep 1; done'",
+            ],
+        },
+    ]
+
+    return app_build_steps + ocis_service + wait_for_ocis
+
+def uploadTracingResult(ctx):
+    return [{
+        "name": "upload-tracing-result",
+        "image": PLUGINS_S3,
+        "pull": "if-not-exists",
+        "settings": {
+            "bucket": {
+                "from_secret": "cache_public_s3_bucket",
+            },
+            "endpoint": {
+                "from_secret": "cache_public_s3_server",
+            },
+            "path_style": True,
+            "source": "test-results/**/*",
+            "strip_prefix": "test-results",
+            "target": "/${DRONE_REPO}/${DRONE_BUILD_NUMBER}/tracing",
+        },
+        "environment": {
+            "AWS_ACCESS_KEY_ID": {
+                "from_secret": "cache_public_s3_access_key",
+            },
+            "AWS_SECRET_ACCESS_KEY": {
+                "from_secret": "cache_public_s3_secret_key",
+            },
+        },
+        "when": {
+            "status": ["failure"],
+        },
+    }]
+
+def logTracingResult(ctx):
+    return [{
+        "name": "log-tracing-result",
+        "image": OC_UBUNTU,
+        "commands": [
+            "cd test-results/",
+            'echo "To see the trace, please open the following link in the console"',
+            'for f in */; do echo "npx playwright show-trace https://cache.owncloud.com/public/${DRONE_REPO}/${DRONE_BUILD_NUMBER}/tracing/$f"trace.zip" \n"; done',
+        ],
+        "when": {
+            "status": ["failure"],
+        },
+    }]
+
+def e2eTests(ctx):
+    e2e_test_steps = []
+
+    for app in APPS_COVERED_E2E:
+        e2e_test_steps.append({
+            "name": app,
+            "image": OC_CI_NODEJS,
+            "commands": [
+                "pnpm exec playwright install chromium",
+                "BASE_URL_OCIS=https://ocis:9200 pnpm test:e2e --project='%s-chromium'" % app,
+            ],
+        })
```

Comment:
you should move chromium step out of the loop. because we can install it once and run e2e tests on available apps.
```suggestion
    e2e_test_steps = [{
        "name": "install-browser",
        "image": OC_CI_NODEJS,
        "commands": [
            "pnpm exec playwright install chromium",
        ]
    }]

    for app in APPS_COVERED_E2E:
        e2e_test_steps.append({
            "name": app,
            "image": OC_CI_NODEJS,
            "commands": [
                "BASE_URL_OCIS=https://ocis:9200 pnpm test:e2e --project='%s-chromium'" % app,
            ],
        })
```